_posts/en/newsletters/2019-04-30-newsletter.md

```diff
@@ -0,0 +1,203 @@
+  sets a dust limit of 546 satoshis as the minimum amount for an output,
+  which seems a peculiar amount. Raghav Sood describes how 546 is three
+  times the minimum cost to create and spend a P2PKH output. A reference
+  is made to a [2013 discussion][dust convo].
+
+- [History of transactions in Lighting Network.]({{bse}}85901) A LN
+  beginner asks how the history of transactions conducted on LN for a
+  user can be saved and how a payer receives a proof of payment. Mark H
+  responds saying the LN wallet would need to save transaction history
+  for a user and provides a nice explanation of how a payment hash
+  provided to a payer results in a payment preimage reveal that serves
+  as proof of payment.
+
+- [How can my private key be revealed if I use the same nonce while
+  generating the signature?]({{bse}}85638) Pieter Wuille provides a
+  thorough answer that, if you’re familiar with the math used in
+  public key cryptography, demonstrates how a private
+  key is revealed in such circumstances.
+
+- [Why do lightning invoices expire?]({{bse}}85981) Rene Pickhardt
+  guesses that the primary reason would be a high
+  volume recipient with relatively low storage capability could run out
+  of storage or memory. An additional reason given is to provide some closure to
+  proceed if a payment is not initiated or completed rather than leave
+  it dangling. David Harding points out that traditional businesses put
+  expiration dates on invoices to avoid an obligation to deliver goods in the
+  future at a previously offered price.
+
+- [Are there still miners or mining pools which refuse to implement
+  SegWit?]({{bse}}86208) Mark Erhardt provides extensive analysis demonstrating
+  that essentially the answer is "no." Only 0.03% non-empty blocks from
+  the past year had no SegWit transactions, and the two primary miners of
+  those few blocks have demonstrated in 2019 that they’re mining blocks
+  with SegWit transactions.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].  Note: unless otherwise noted, all merges described
+for Bitcoin Core are to its master development branch; some may also be
+backported to its pending release.*
+
+- [Bitcoin Core #14039][] causes Bitcoin Core to reject transactions
+  that use the segwit-style extended transaction encoding when they
+  don't include any segwit inputs.  The extended transaction encoding
+  includes a segwit marker, a segwit flag, and witness data fields.
+  Signatures included in legacy inputs don't commit to these fields, so
+  adding the fields to a transaction produces a (small) waste of
+  bandwidth in a transaction consisting entirely of legacy inputs.  For
+  this reason, [BIP144][] specified that non-segwit transactions should
```

Comment:
I think the terms 'segwit transaction' and 'non-segwit transaction' are not explicit enough, and could easily be interpreted as meaning a transaction that sends to a segwit output. I think it's better to always be explicit and say "a transaction that spends a segwit output".